### PR TITLE
 chore(tools): add prettier staging hook

### PR DESCRIPTION
**Pull request by tkrugg**
Nothing more to say than what's in the title 😄

### CODE REVIEW

**Comment by tkrugg** [comment]
I thought every commit would be clean this way and it's what I was used to do.
It adds to the commit time for sure, but since it's only checking the staged files it's rather quick.
I'll switch to pre-push if that's what we prefer

**Comment by samouss** [comment]
Ok I’m fine with it.

File: package.json [hunk 1]
```diff
@@ -125,8 +136,10 @@
     "eslint-plugin-import": "2.14.0",
     "eslint-plugin-prettier": "3.0.1",
     "file-loader": "1.1.11",
+    "husky": "^1.3.1",
```
Comment (francoischalifour) [inline package.json]: Should be pinned.

File: package.json [hunk 2]
```diff
@@ -125,8 +136,10 @@
...
     "jest": "23.6.0",
     "jest-preset-angular": "6.0.2",
+    "lint-staged": "^8.1.4",
```
Comment (francoischalifour) [inline package.json]: Should be pinned.

**Comment by tkrugg** [comment]
@francoischalifour 
it appears that it does, it's stashing the changes that are not added, runs the linters, restores the stash.

image

Please tell us more about the "bad experiences with pre-commit" and "the flaws" of each solution. I don't see any atm.

**Comment by francoischalifour** [comment]
On top of my head, when using anything else that `precise-commits`, it runs Prettier on the whole file. You can see the table differences on the [`precise-commits` repo](https://github.com/nrwl/precise-commits#why-precise-commits). That's unfortunate.

These hooks also slow down the commit flow but that's a tradeoff that we're OK to accept for the benefit it provides IMO.

**Pull request closed without merging**